### DISCUSSION TITLE
Build fails

### DISCUSSION THREAD

**Discussion by gsg-git** [osm-search/Nominatim#3191] [Q&A]
Hello!
I installed every requirement and when i  let the build run it fails with missing applications:
behave, phpunit, phpcs, pylint, pytest
I tried the Git and newest Stable Version.
Ubuntu Server 22.04.3 LTS (GNU/Linux 5.15.0-83-generic x86_64)
Specs:
 processor      13th Gen Intel(R) Core(TM) i5-1360
memory         128GiB System Memory (4*32GiB DIMM Synchronous 4200 MHz (0.2 ns))

Message:
`-- Using PHP binary /usr/bin/php
CMake Warning at CMakeLists.txt:142 (message):
  behave not found.  BDD tests disabled.

CMake Warning at CMakeLists.txt:151 (message):
  phpunit not found.  PHP unit tests disabled.

CMake Warning at CMakeLists.txt:160 (message):
  phpcs not found.  PHP linting tests disabled.

CMake Warning at CMakeLists.txt:169 (message):
  pylint not found.  Python linting tests disabled.

CMake Warning at CMakeLists.txt:178 (message):
  pytest not found.  Python tests disabled.

-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
NLOHMANN_INCLUDE_DIR
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/src
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/src
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/src
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/src
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/src
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/src
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/src
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/src
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/docs
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/docs
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/docs
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/docs
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/docs
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/docs
   used as include directory in directory /home/nominatim/Nominatim/osm2pgsql/docs

#!!Here comes the Errors!!#

**CMake Error in osm2pgsql/CMakeLists.txt:
  Found relative path while evaluating include directories of "osm2pgsql":

    "NLOHMANN_INCLUDE_DIR-NOTFOUND"

CMake Error in osm2pgsql/src/CMakeLists.txt:
  Found relative path while evaluating include directories of
  "osm2pgsql_lib":

    "NLOHMANN_INCLUDE_DIR-NOTFOUND"**

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.

`

I have installed them with:

`pip install behave --user
pip install phpunit --user 
sudo apt install php-cli \
                 php-json \
                 php-mbstring \
                 php-xml \
                 php-pcov \
                 php-xdebug   
apt install php-phpcs
pip install pylint --user
pip install pytest --user`

But even after this it fails the same.
even when i list with php - there is for eg. no phpcs - even after installing via composer https://github.com/squizlabs/PHP_CodeSniffer.

What could be the problem?

Thank you in advance!

**Comment by lonvia** [comment]
Please make sure you use the right installation instructions for what you are installing. If you are installing from master, then the instructions are https://nominatim.org/release-docs/develop/admin/Installation/ For the newest stable version 4.3 it is https://nominatim.org/release-docs/4.3/admin/Installation/ Both mention nlohmann-json as dependency.
